databricks/koalas/indexes.py

```diff
@@ -571,6 +571,32 @@ def __getattr__(self, item: str) -> Any:
     def rename(self, name, inplace=False):
         raise NotImplementedError()
 
+    @property
+    def levels(self) -> list:
+        """
+        Names of index columns in list.
+
+        Examples:
+        --------
+        >>> mi = pd.MultiIndex.from_arrays((list('abc'), list('def')))
+        >>> mi.names = ['level_1', 'level_2']
+        >>> kdf = ks.DataFrame({'a': [1, 2, 3]}, index=mi)
+        >>> kdf.index.levels
+        [['a', 'b', 'c'], ['d', 'e', 'f']]
+
+        >>> mi = pd.MultiIndex.from_arrays((list('bac'), list('fee')))
+        >>> mi.names = ['level_1', 'level_2']
+        >>> kdf = ks.DataFrame({'a': [1, 2, 3]}, index=mi)
+        >>> kdf.index.levels
+        [['a', 'b', 'c'], ['e', 'e', 'f']]
+        """
+        idx_cols = self._kdf._internal.index_columns
+        sdf = self._kdf._sdf.select(idx_cols).dropDuplicates()
+
+        # use sorting is because pandas doesn't care the appearance order of level
+        # names, so e.g. if ['b', 'd', 'a'] will return as ['a', 'b', 'd']
+        return [sorted([row[col] for row in sdf.collect()]) for col in idx_cols]
```

Comment:
This invokes Spark job multiple times.
How about using `F.collect_set`? And then we don't need `dropDuplicates`.

```py
self._kdf._sdf.select([F.collect_set(scol) for scol in self._kdf._internal.index_scols]).collect()
```

Reply:
nice one!